test/junit/scala/collection/mutable/BitSetTest.scala

```diff
@@ -48,4 +48,14 @@ class BitSetTest {
     val bsFromEmptyBitMaskNoCopy = BitSet.fromBitMaskNoCopy(Array.empty[Long])
     assert(bsFromEmptyBitMaskNoCopy.add(0))
   }
+
+  @Test def strawman_508: Unit = {
+    val m = BitSet(1)
+    assert(m.map(i=>i.toLong) == TreeSet(1L))
+    assert(m.map(i=>i + 1) == BitSet(2))
```

Comment:
The test is not good because `BitSet(2) == TreeSet(2)`. We should test that the runtime type is effectively `BitSet` (and same line 54, we should test that the runtime type is `TreeSet`). We can do that like so:

~~~ scala
assert(m.map(i => i.toLong).isInstanceOf[TreeSet[Long]])
assert(m.map(i => i + 1).isInstanceOf[BitSet])
~~~

Reply:
I updated the test as suggested.